File: QsWebSoft/Beta3/W_TreeView.win.js
///<reference path="/Scripts/ExtPB_Net.js" />
///<reference path="/Scripts/MyExtPB.js" />
///<reference path="/Beta3/ExtPB_Demo.js" />

function W_TreeView() {

    var self = this;
    var requestor = new PBWindow();
    var tv_1 = new TV_Category(requestor);
    var dw_1 = new DataWindow(requestor);
    var tb_1 = new ToolStrip(requestor);
    var ds_Categories = new DataStore(requestor);

    if (typeof (this.bind) == "function") var baseBind = this.bind;
    this.bind = function (win) {
        if (typeof (baseBind) == "function") baseBind(win);

        if (requestor == null) requestor = new PBWindow();
        requestor.Attach(win);
        if (tv_1 == null) tv_1 = new TV_Category(requestor);
        tv_1.Attach(win.Control("tv_1"));
        if (dw_1 == null) dw_1 = new DataWindow(requestor);
        dw_1.Attach(win.Control("dw_1"));
        if (tb_1 == null) tb_1 = new ToolStrip(requestor);
        tb_1.Attach(win.Control("tb_1"));
        if (ds_Categories == null) ds_Categories = new DataStore(requestor);
        ds_Categories.Attach(win.Control("ds_Categories"));

        

    }

    if (typeof (this.unbind) == "function") var baseUnBind = this.unbind;
    this.unbind = function () {
        if (typeof (this.Release) == "function") this.Release();
        if (typeof (baseUnBind) == "function") baseUnBind();

        tv_1.Detach(); tv_1 = null;
        dw_1.Detach(); dw_1 = null;
        tb_1.Detach(); tb_1 = null;
        ds_Categories.Detach(); ds_Categories = null;
        requestor.Detach(); requestor = null;

        setTimeout(CollectGarbage, 100);
    }

    //以上代码由设计器自动生成,请不要手工修改
    //ExtPB.Net JavaScript End  设计器查找标志,请不要删除或修改
    
}

File: QsWebSoft/Beta3/W_TreeView.win.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Text;
using System.Web;
using System.Data;
using System.Data.SqlClient;

using Sybase.DataWindow;
using TXSoft.Common;
using TXSoft.ExtPB;
using TXSoft.DataStore;


namespace QsWebSoft.Beta3
{
    public partial class W_TreeView
    {
        public W_TreeView()
        {

        }
        public override void OnLoad()
        {

           

            base.OnLoad();

            //检索分类
            ds_Categories.SetTransaction(this.AdoTransaction);
            ds_Categories.Retrieve();

            //检索第一个分类的产品资源
            if (ds_Categories.RowCount > 0)
            {
                dw_1.SetTransaction(this.AdoTransaction);
                dw_1.Retrieve(ds_Categories.GetItemInt32(1, "categoryid")); 
            }
            
            //行选择服务
            dw_1.RowSelection = RowSelectionStyle.Single; 


            
            //注意，树控件的JavaScript的类对象不是缺省的ExtPB.Net TreeView
            //而是自定义的一个JavaScript类对象, TV_Category,专门用于生成产品分类的树结构

            //树控件邦定的DataStore对象名称
            tv_1.StringData1 = "ds_Categories";


            //树控件关联的产品列表数据窗口对象名称
            tv_1.StringData2 = "dw_1";

            //注册JS文件
            this.RegisterClientScriptInclude("TV_Category", "/Beta3/TV_Category.js");

        }
    }
}
